src/history.cpp

```diff
@@ -1453,7 +1453,10 @@ bool history_t::save_internal_via_rewrite()
                 if (wstat(new_name, &sbuf) >= 0)
                 {
                    /* Success */
-                    fchown(out_fd, sbuf.st_uid, sbuf.st_gid);
+                    if (0 > fchown(out_fd, sbuf.st_uid, sbuf.st_gid))
```

Comment:
That idiom is used in exactly three places, not counting your change, and is bollocks. Please change both. Leave the other two in *fish.cpp* and *common.cpp* for another day. I'm going to work with the other fish maintainers to implement a more consistent style but that's a battle for another day.